core/src/main/java/org/apache/calcite/rel/metadata/RelMdCollation.java

```diff
@@ -275,9 +276,10 @@ private RelMdCollation() {}
    * from each of its windows. Assuming (quite reasonably) that the
    * implementation does not re-order its input rows, then any collations of its
    * input are preserved. */
+  @Deprecated // to be removed before 2.0
```

Comment:
Why do you mark it as deprecated?

Reply:
I suppose users would use this class since it provides public util methods for computing collations.

Reply:
And the method RelMdCollation#window will always return empty collations -- this is covered by the collation method for general RelNode, so in fact this method can be removed.